lucene/expressions/src/java/org/apache/lucene/expressions/js/JavascriptCompiler.java

```diff
@@ -206,6 +247,12 @@ private Expression compileExpression(ClassLoader parent) throws ParseException {
       throw new IllegalStateException("An internal error occurred attempting to compile the expression (" + sourceText + ").", exception);
     }
   }
+  
```

Comment:
I wouldn't do this, to be honest. Rethrow Error and RuntimeException subclasses as they were, wrap anything else in a runtime exception with an appropriate caused-by delegate. Makes it easier to reason about the code should something happen.

Reply:
I changed it. The way how to do this is correct as it only intends to work around a problem of type inference in javac. For a methodhandle we know for sure, which exceptions are thrown (it's different to core reflection, where everything is wrapped in InvocationTargetException).

I moved the whole thing to a separate method and I catched and rethrow all "known" Exceptions (IllegalAccessException, RuntimeException, Error), all others can't happen, so it wraps with AssertionError.